stan/math/prim/functor/apply_vector_unary.hpp

```diff
@@ -151,13 +190,30 @@ struct apply_vector_unary<T, require_std_vector_vt<is_container, T>> {
   template <typename F>
   static inline auto apply(const T& x, const F& f) {
     size_t x_size = x.size();
-    using T_return = decltype(apply_vector_unary<T_vt>::apply(x[0], f));
+    using T_return
+        = plain_type_t<decltype(apply_vector_unary<T_vt>::apply(x[0], f))>;
     std::vector<T_return> result(x_size);
     for (size_t i = 0; i < x_size; ++i)
-      result[i] = apply_vector_unary<T_vt>::apply(x[i], f);
+      result[i] = apply_vector_unary<T_vt>::apply_no_holder(x[i], f);
     return result;
   }
 
+  /**
+   * Member function for applying a functor to each container in an std::vector
+   * and subsequently returning an std::vector of containers.
+   *
+   * @tparam T Type of argument to which functor is applied.
+   * @tparam F Type of functor to apply.
+   * @param x std::vector of containers to which operation is applied.
+   * @param f functor to apply to vector input.
+   * @return std::vector of containers with result of applying functor to
+   *         input.
+   */
+  template <typename F>
+  static inline auto apply_no_holder(const T& x, const F& f) {
+    return apply(x, f);
```

Comment:
Does this get used anywhere?

Reply:
Yes, in `std::vector` specialization of `apply_vector_unary`.